app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryEffectHandler.kt

```diff
@@ -66,4 +72,12 @@ class BloodSugarEntryEffectHandler(
 
   private fun getYear(date: LocalDate): String =
       date.year.toString().substring(startIndex = 2, endIndex = 4)
+
+  private fun showDateValidationError(result: UserInputDateValidator.Result) {
+    when (result) {
+      InvalidPattern -> ui.showInvalidDateError()
+      DateIsInFuture -> ui.showDateIsInFutureError()
+      is Valid -> throw IllegalStateException("Date validation error cannot be $result")
```

Comment:
Since this `when` block does not return anything, adding a new type to `UserInputDateValidator.Result` will result in no effect.

Should we change this to an `else` instead so that the app will fail during development? The other option is to use the `.exhaustive()` extension function to ensure that we have added all types to this block.

```kotlin
when (result) {
  InvalidPattern -> ui.showInvalidDateError()
  DateIsInFuture -> ui.showDateIsInFutureError()
  is Valid -> throw IllegalStateException("Date validation error cannot be $result")
}.exhaustive()
```

Reply:
Added `.exhaustive()` extension. Forgot Kotlin doesn't fail build for missing branches in `when` block when we don't return it.